x-pack/plugins/security_solution/server/usage/collector.ts

```diff
@@ -308,6 +309,1604 @@ export const registerCollector: RegisterCollector = ({
+                top_failed: {
+                  type: 'array',
+                  items: {
+                    message: {
+                      type: 'keyword',
+                      _meta: { description: 'Failed rule message' },
+                    },
+                    count: {
+                      type: 'long',
+                      _meta: { description: 'Number of times the message occurred' },
+                    },
+                  },
+                },
+                partial_failure: {
+                  type: 'long',
+                  _meta: { description: 'The number of partial failure rules' },
+                },
+                top_partial_failure: {
+                  type: 'array',
+                  items: {
+                    message: {
+                      type: 'keyword',
+                      _meta: { description: 'Failed rule message' },
+                    },
+                    count: {
+                      type: 'long',
+                      _meta: { description: 'Number of times the message occurred' },
+                    },
+                  },
+                },
+                succeeded: {
+                  type: 'long',
+                  _meta: { description: 'The number of successful rules' },
+                },
+                index_duration: {
+                  max: {
+                    type: 'float',
+                    _meta: { description: 'The max duration' },
+                  },
+                  avg: {
+                    type: 'float',
+                    _meta: { description: 'The avg duration' },
+                  },
+                  min: {
+                    type: 'float',
+                    _meta: { description: 'The min duration' },
+                  },
+                },
+                search_duration: {
+                  max: {
+                    type: 'float',
+                    _meta: { description: 'The max duration' },
+                  },
+                  avg: {
+                    type: 'float',
+                    _meta: { description: 'The avg duration' },
+                  },
+                  min: {
+                    type: 'float',
+                    _meta: { description: 'The min duration' },
+                  },
+                },
+                gap_duration: {
+                  max: {
+                    type: 'float',
+                    _meta: { description: 'The max duration' },
+                  },
+                  avg: {
+                    type: 'float',
+                    _meta: { description: 'The avg duration' },
+                  },
+                  min: {
+                    type: 'float',
+                    _meta: { description: 'The min duration' },
+                  },
+                },
+                gap_count: {
+                  type: 'long',
+                  _meta: { description: 'The count of gaps' },
+                },
+              },
+              savedQuery: {
```

Comment:
This rule type should've been deprecated by now, but thank you for supporting it as it's still possible to create this type. Will be good to start getting status metrics telemetry on it! 🙂

Reply:
That said, doesn't look like we report `detection_rule_usage` data on this sub-type? (ignore ml highlight, re-using same image from before).

<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/160668114-97d00f33-be00-4ded-9d93-ebcbb11c7819.png" />
</p>

Reply:
Yes, I don't know much about the savedQuery rule to be completely honest. But it sounds like it's deprecated. I can keep it or remove it if you want me to. I will at least convert it to snake case with the rest.

Reply:
Yeah, let's keep it so we can get the data -- I just tested creating a custom query rule from a saved query and it gets created as `saved_query`, so might as well get these metrics back so we can make an informed decision if/when deprecating.


<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/160694400-a6e2b1eb-acb5-46fb-835b-f8611df45f74.png" />
</p>

Reply:
Agreed. It should be plumbed for the other metrics but it has not been. I will keep it for these metrics.